exp/api/v1alpha3/azuremanagedcontrolplane_types.go

```diff
@@ -96,15 +96,15 @@ type AADProfile struct {
 
 // ManagedControlPlaneVirtualNetwork describes a virtual network required to provision AKS clusters.
 type ManagedControlPlaneVirtualNetwork struct {
-	Name      string                    `json:"name"`
-	CIDRBlock string                    `json:"cidrBlock"`
-	Subnet    ManagedControlPlaneSubnet `json:"subnet,omitempty"`
+	Name       string                      `json:"name"`
+	CIDRBlocks []string                    `json:"cidrBlocks"`
```

Comment:
> how can we differentiate between {}string[] and nil

You should probably have a non-generated/handwritten func for these special cases, since you changed a `string` field to `[]string`.

I ran into similar issues with conversion tests before, take a look here for example, maybe setting the field explicitly to `nil` similarly to this will help https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/api/v1alpha3/azuremachine_conversion.go#L41-L46

Reply:
> You should probably have a non-generated/handwritten func for these special cases, since you changed a string field to []string.

Basically, when there is a conversion from an old object which had `CIDRBlock` set to an empty string, I guess that should be converted to `CIDRBlocks` set to an empty array in the new object.